Sentinel/Classes/Core/Internal/SentinelInternal.swift

```diff
@@ -26,37 +28,75 @@ extension Sentinel {
 }
 
 private extension Sentinel {
-    func createTabBarControllers(with tools: [Tool], viewController: UIViewController) -> [UIViewController] {
-        var tabBarControllers = tools.filter { tool in
-            return tool.type == .application ||
-            tool.type == .permissions ||
-            tool.type == .performance
-        }.map { $0.createViewController(on: viewController) }
-
-        let toolItems = tools.filter { tool in
-            return tool.type != .application &&
-            tool.type != .permissions &&
-            tool.type != .performance
-        }
-
-        let toolsViewController = createToolsController(with: toolItems, viewController: viewController)
-        let deviceViewController = createDeviceViewController()
-
-        tabBarControllers.insert(deviceViewController, at: 0)
-        tabBarControllers.insert(toolsViewController, at: 2)
+    func createTabBarControllers(with tools: [Tool], optionSwitchItems: [OptionSwitchItem], viewController: UIViewController) -> [UIViewController] {
+        var tabBarControllers: [UIViewController] = []
+        tabBarControllers.append(createDeviceViewController())
+        tabBarControllers.append(createApplicationViewController())
+        tabBarControllers.append(createToolsController(with: tools, viewController: viewController))
+        tabBarControllers.append(createPreferencesViewController(optionSwitchItems))
+        tabBarControllers.append(createPerformanceViewController())
         return tabBarControllers
     }
 
     func createToolsController(with tools: [Tool], viewController: UIViewController) -> UIViewController {
```

Comment:
```suggestion
    func createToolsController(with tools: [Tool]) -> UIViewController {
```
View controller doesn't seem to be used any more :)

Reply:
Nice catch 😄